### PR TITLE
Add Razer BlackWidow 2019 support

### LINKED ISSUE #844: Support for BlackWidow (2019)
### Device Info

* Name: Razer BlackWidow
* Product Number: RZ03-02860100-R3M1
* Model Number: RZ03-0286
### Device Information
```
Bus 003 Device 002: ID 1532:0241 Razer USA, Ltd
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x0241
  bcdDevice            2.00
  iManufacturer           1
  iProduct                2
  iSerial                 3
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors:
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0
      bInterfaceProtocol      1 Keyboard
      iInterface              0
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     179
         Report Descriptors:
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0016  1x 22 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0
      bInterfaceProtocol      2 Mouse
      iInterface              0
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors:
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1

```

```
total 0
drwxr-xr-x 2 root root 340 mai   20 08:31 .
drwxr-xr-x 4 root root 760 mai   20 08:31 ..
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-DAC_FOR_USB_FOCAL_XS-event-if02 -> ../event18
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Logitech_G502_HERO_Gaming_Mouse_0E67385B3538-event-if01 -> ../event16
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Logitech_G502_HERO_Gaming_Mouse_0E67385B3538-event-mouse -> ../event14
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Logitech_G502_HERO_Gaming_Mouse_0E67385B3538-if01-event-kbd -> ../event15
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Logitech_G502_HERO_Gaming_Mouse_0E67385B3538-mouse -> ../mouse2
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Razer_Razer_BlackWidow-event-if01 -> ../event10
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_BlackWidow-event-kbd -> ../event8
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_BlackWidow-if01-event-kbd -> ../event9
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Razer_Razer_BlackWidow-if02-event-mouse -> ../event13
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_BlackWidow-if02-mouse -> ../mouse1
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Razer_Razer_Goliathus_Chroma-event-mouse -> ../event26
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_Goliathus_Chroma-if01-event-kbd -> ../event4
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_Goliathus_Chroma-if02-event-kbd -> ../event5
lrwxrwxrwx 1 root root   9 mai   20 08:31 usb-Razer_Razer_Goliathus_Chroma-mouse -> ../mouse0
lrwxrwxrwx 1 root root  10 mai   20 08:31 usb-Razer_Razer_Kraken_Tournament_Edition_000000000000000000000000-event-if05 -> ../event20

```

Thanks!

### PR DESCRIPTION
**Pull request by frikiluser**
Close: #844

It also fixes a recent typo in "install_files/udev/99-razer.rules". Missing leading 0.

I've tested all examples with successful results.
"advanced_effect.py" shows a big variety of effects, probably all available: breath_dual breath_random breath_single reactive spectrum static wave

Probably isn't perfect, because I haven't tested "official" driver.

### CODE REVIEW

File: daemon/openrazer_daemon/hardware/keyboards.py
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/keyboards.py]: Please use https://assets.razerzone.com/eeimages/support/products/1501/1501-blackwidow2019.png as picture here

**Comment by frikiluser** [comment]
@z3ntu I've updated the URL, as requested

https://github.com/openrazer/openrazer/issues/844#issuecomment-503476183
In the issue thread @asyd mentioned a problem related to Fn key. Code reviewers must be aware.

**Review by z3ntu — APPROVED** [review]
LGTM

**Comment by z3ntu** [comment]
Btw your git committer email (antoni at friki dot cat) is not registered with your GitHub account, so the contribution isn't really connected to your github account.

**Comment by asyd** [comment]
Indeed with this branch, `xev` reports 

```
KeyRelease event, serial 40, synthetic NO, window 0x7800001,
    root 0x1e3, subw 0x0, time 1549718, (1899,1658), root:(1899,1707),
    state 0x10, keycode 72 (keysym 0xffc3, F6), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```

When I press `fn+F6`, so it's the same code than `F6` key.

I tried to comment the line `    case USB_DEVICE_ID_RAZER_BLACKWIDOW_2019:` in `razer_raw_event` function but I have the same issue.

**Comment by frikiluser** [comment]
@asyd It works for me with the new changes at "razer_raw_event".

I doesn't know what it means, but... size == 22, not 16.

**Comment by asyd** [comment]
My bad, I missed them. I confirm it's working!

**Comment by frikiluser** [comment]
I think it's a final version now. The prev only works for BlackWidow, hehe

@z3ntu The pull request is a minimal impact version.
At branch "raw_proposal" I've modified slightly the aborting conditions. It can be done in a different manners. I don't have the knowledge beyond these code lines, so I can't write proper code. My proposal is add some consistence between different checks :-)

File: driver/razerkbd_driver.c
Comment (z3ntu) [inline driver/razerkbd_driver.c]: Why was this change done (plus the few lines above)?

**Review by z3ntu — CHANGES_REQUESTED** [review]
See comment, LGTM otherwise

File: driver/razerkbd_driver.c
Comment (frikiluser) [reply driver/razerkbd_driver.c]: Hi,

I've observed size=22 when Fn key is pressed on this keyboard.
So, the changes requested will break Fn key again.

Let me know if I can help with some more debug

Edit:
@z3ntu If you prefer, I can revert my changes on that function and open an issue after merge pull request. Most hardware support will be provided.
IMHO Fn key is a minor issue.

Comment (z3ntu) [reply driver/razerkbd_driver.c]: Yes please do so.

Comment (frikiluser) [reply driver/razerkbd_driver.c]: Done.
Regards,

Comment (asyd) [reply driver/razerkbd_driver.c]: as owner of a blackwidow 2019 I disagree about "fn key is a minor issue" ;p

But off course I can understand the workflow!

**Review by z3ntu — APPROVED** [review]
LGTM

**Comment by z3ntu** [comment]
@frikiluser Please open a PR with the fn key stuff so it can be reviewed there

**Comment by asyd** [comment]
Sounds like the merge about FN key is missing. Just bought a new box and install openrazer from debian repository (version `2.6.0`) and FN key is broken. I'll try to check the piece of code involved

**Pull request merged**
